plugins/catalog-react/src/components/EntityRefLink/humanize.ts

```diff
@@ -65,3 +66,36 @@ export function humanizeEntityRef(
       : kind;
   return `${kind ? `${kind}:` : ''}${namespace ? `${namespace}/` : ''}${name}`;
 }
+
+/**
+ * Convert an entity to its more readable name if available.
+ *
+ * If an entity is either User or Group, this will be its `spec.profile.displayName`.
+ * Otherwise, this is `metadata.title`.
+ *
+ * If neither of those are found or populated, fallback to `humanizeEntityRef`.
+ *
+ * @param entity - Entity to convert.
+ * @param opts - If entity readable name is not available, opts will be used to specify humanizeEntityRef options.
+ * @returns Readable name, defaults to unique identifier.
+ *
+ * @public
+ */
+export function humanizeEntity(
+  entity: Entity,
+  opts?: {
+    defaultKind?: string;
+    defaultNamespace?: string | false;
+  },
+) {
+  let title: string | undefined = undefined;
+  switch (entity.kind) {
+    case 'User':
+    case 'Group':
+      title = (entity as UserEntity).spec?.profile?.displayName;
```

Comment:
How about if there is no displayName but there is metadata.title for example for group? I think this is very valid case

Reply:
Happy to add it as a fallback but I don't think it's something that is supported out of the box.

I was basing it off of the examples in https://backstage.io/docs/features/software-catalog/descriptor-format#kind-group and https://backstage.io/docs/features/software-catalog/descriptor-format#kind-user as well as https://github.com/sennyeya/backstage/blob/eebbbfc5168ed32223e95ed3193d4e1a31fa18bb/packages/catalog-model/examples/acme/team-a-group.yaml#L38. The `metadata.title` doesn't seem to be populated and the expected value is the `spec.profile.displayName` instead. 

Taking a look at existing processors, it looks like that's the expected pattern as well. 
- https://github.com/sennyeya/backstage/blob/eebbbfc5168ed32223e95ed3193d4e1a31fa18bb/plugins/catalog-backend-module-gitlab/src/providers/GitlabOrgDiscoveryEntityProvider.ts#L348
- https://github.com/sennyeya/backstage/blob/eebbbfc5168ed32223e95ed3193d4e1a31fa18bb/plugins/catalog-backend-module-ldap/src/ldap/read.ts#L86.

Reply:
Might we want to do both, ie use `get` for more typesafety but still do a switch to prevent picking up `spec.profile.displayName` where it isn't part of the spec for the specified kind? 

Seems like this would pick up `spec.profile.displayName` for non-User/Group entities which may not be desired.